### PR TITLE
T-Vault login using AAD credentials

### PR DESCRIPTION
**Pull request by sarathmohan4**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/VaultAuthService.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ else if (HttpStatus.INTERNAL_SERVER_ERROR.equals(response.getHttpstatus())) {
 	 * @param access
 	 * @return
 	 */
-	private Map<String,Object> filterDuplicateSafePermissions(Map<String,Object> access) {
+	public Map<String,Object> filterDuplicateSafePermissions(Map<String,Object> access) {
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/VaultAuthService.java:121]: Can this be refactored to an Util Class?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+public class OidcAuthService {
+
+	@Autowired
+	private RequestProcessor reqProcessor;
+	
+	@Autowired
+	private AuthorizationUtils authorizationUtils;
+
+	@Autowired
+	private VaultAuthService vaultAuthService;
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java:53]: We need to be careful in using a Service class in another?
If it is required, the dependent method(s) can be moved to an Util class/new class (filterDuplicateSafePermissions)

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	}
+
+	/**
+	 * To get vault token with OIDC callback state and code.
+	 * @param state
+	 * @param code
+	 * @return
+	 */
+	public ResponseEntity<String> processCallback(String state, String code) {
+        String token = tokenUtils.getSelfServiceToken();
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java:101]: Why is this called for every time? Is it not returned by TokenValidationFilter?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        String pathStr = "?code="+code+"&state="+state;
+		Response response = reqProcessor.process("/auth/oidc/oidc/callback","{\"path\":\""+pathStr+"\"}",token);
+		if(HttpStatus.OK.equals(response.getHttpstatus())){
+			Map<String, Object> responseMap = null;
+			try {
+				responseMap = new ObjectMapper().readValue(response.getResponse(), new TypeReference<Map<String, Object>>(){});
+			} catch (IOException e) {
+				log.error(JSONUtil.getJSON(ImmutableMap.<String, String>builder().
+						put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER).toString()).
+						put(LogMessage.ACTION, "processCallback").
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java:112]: What is meant by "processCallback"?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/VaultAuthService.java [hunk 5]
```diff
@@ -151,7 +151,7 @@ else if (HttpStatus.INTERNAL_SERVER_ERROR.equals(response.getHttpstatus())) {
 	 * @param access
 	 * @return
 	 */
-	private Map<String,Object> filterDuplicateSvcaccPermissions(Map<String,Object> access) {
+	public Map<String,Object> filterDuplicateSvcaccPermissions(Map<String,Object> access) {
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/VaultAuthService.java:154]: Again can this be refactored to an Util Class?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/OidcAuthControllerV2.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+@CrossOrigin
+@Api(description = "Manage OIDC Authentication", position = 19)
+public class OidcAuthControllerV2 {
+
+    @Autowired
+    private OidcAuthService oidcAuthService;
+
+    /**
+     * Login to TVault
+     * @returnC
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/OidcAuthControllerV2.java:40]: Additional Character C?

Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/OidcAuthControllerV2.java:40]: Removed

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/VaultAuthService.java (hunk 5 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/VaultAuthService.java:154]: Moved filter functions to ControllerUtil

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java (hunk 4 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java:112]: This meant to process the oidc callback. I have renamed this to processOidcCallback

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java (hunk 3 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/OidcAuthService.java:101]: As this api will not take any token, tokenValidation filter will not give the selfservice token as there is no userdetails

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/VaultAuthService.java (hunk 1 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/VaultAuthService.java:121]: Moved filter functions to ControllerUtil

**Pull request merged**